video/out/drm_atomic.c

```diff
@@ -382,8 +383,10 @@ bool drm_atomic_save_old_state(struct drm_atomic_context *ctx)
 
     if (!drm_atomic_save_plane_state(ctx->osd_plane, &ctx->old_state.osd_plane))
         ret = false;
-    if (!drm_atomic_save_plane_state(ctx->video_plane, &ctx->old_state.video_plane))
-        ret = false;
+    if (ctx->video_plane) {
+        if (!drm_atomic_save_plane_state(ctx->video_plane, &ctx->old_state.video_plane))
```

Comment:
Maybe
```c
if (ctx->video_plane &&
    !drm_atomic_save_plane_state(ctx->video_plane, &ctx->old_state.video_plane)) {
    ret = false;
}
```
?

Reply:
This also in the future could be improved with a macro that checks the variable and does the atomic plane state save.

Reply:
I could also move the NULL-check into drm_atomic_save_plane_state. Would that be good?